Remove carousel max height add carousel scroll down button, add carousel scroll down button, upgrade jquery, move js libs

adamhb123 opened the pull request:
Removes the max-height styling for the scrolling carousel.
 
 Adds the carousel scroll down button, which solves the design issue
 of images taking up the whole page (which upon first viewing might
 lead someone to believe that is all the page contains) by providing
 an indicator+button that scrolls to reveal the rest of the page.
 
 Upgrades jquery from slim to full. This is required so we can use
 animate for the smooth scrolling functionality.
 
 Moves all javascript libraries from the footer to header, if this
 is undesirable for any particular reason (I really didn't see any
 issues), just say so and I'll pull it from the PR.
 
 Resolves #280

Lontronix commented:
@adamhb123 this is looking good so far but please adjust the symbol that is used to scroll the page down. Ideally a vector that's a bit wider. I think the V looks out of place

Dr-N0 commented:
I agree, also would like to make sure that it scrolls well

adamhb123 commented:
Too lazy to add video, but new symbol is used, scrolling is smoother, etc...

adamhb123 commented:
Going to re-do some of this to use a vector in place of the character meme. Better to do this in the long-run anyways

Mstrodl commented:
Brewer...

You know you can just use like... [window.scrollTo](https://developer.mozilla.org/en-US/docs/Web/API/Window/scrollTo) with [behavior: "smooth"](https://developer.mozilla.org/en-US/docs/Web/API/ScrollToOptions/behavior), right? This is [part of] why I hate jQuery :(

adamhb123 commented:
> Brewer...
> 
> You know you can just use like... [window.scrollTo](https://developer.mozilla.org/en-US/docs/Web/API/Window/scrollTo) with [behavior: "smooth"](https://developer.mozilla.org/en-US/docs/Web/API/ScrollToOptions/behavior), right? This is [part of] why I hate jQuery :(

Strodl...

We already use the jQuery library to begin with, so I don't see a significant incentive **not** to use jQuery's animate function, especially since it allows us to specify duration as well as provides the ability to select an easing function that isn't just linear vs smooth. Also, the animate function allows for a step callback so we can update where the final "landing-point" of the animation will lie (as the size of the navbar is responsive, and we want to align the bottom of the navbar to bottom of the slick carousel.

Mstrodl commented:
@adamhb123 

> We already use the jQuery library to begin with, so I don't see a significant incentive not to use jQuery's animate function

The reason why I bring this up is because we're switching from the slim build which ideally I think we should avoid. I think in the case of updating the landing point, you can use the `element.scrollTo` thing to achieve what you want. You certainly get less granular control over scrolling, but I'm not sure that we really need to fine-tun it that much.

Either way, at the end of the day it's your PR, but I think we should optimize for smaller filesizes.

adamhb123 commented:
I see your points but the difference between the min and slim builds is like 6k, I really don't think that is a big deal in the modern era. 

If you were to have mentioned that the jquery-ui library I added is pretty sizeable (~253kb), then that would be more reasonable, but again, this is something a normal user will only have to load once - and probably already has cached anyways - so again, I don't see a big deal regarding performance here. Even if you're on dialup (probably ~40kbps) it would take some-odd seconds maximum to load the library, and that wait time would decrease significantly for all future visits (assuming the user isn't clearing their cache everytime they visit). Also, I added this library to fulfill @Dr-N0 's request for a smoother scrolling animation, as it is required for the advanced easing functions. 

I feel that the use (and upgrade) of jQuery and the ui library is justified.

The "jquery bad" footgun comment just seems to be a platitude, and doesn't really contribute anything to the discussion.

adamhb123 commented:
In other news I will create and add the new arrow svg to the site sometime this week...very busy times in Ohio

Dr-N0 commented:
> I think it looks fine the way it currently functions. The video is old so if you did not test this locally you aren't seeing the latest changes.

Pulled the branch and things look fine. Mine was outdated
